### PR TITLE
cert collection: rewrite bootstrap IP in secret details

### PR DESCRIPTION
**Pull request by vrutkovs**
etcd leaves bootstrap certificates around, which needs to be anonymized in RewriteNodeIPs. This function is extended to accept bootstrap IP, fetched from an annotation in `etcd-endpoints-*` configmap in `openshift-etcd` namespace.

Alternative to https://github.com/openshift/library-go/pull/1723 and https://github.com/openshift/library-go/pull/1717. Tested in https://github.com/openshift/origin/pull/28657

### CODE REVIEW

**Comment by vrutkovs** [comment]
Tested in https://github.com/openshift/origin/pull/28657:
```
      {
        "secretLocation": {
          "Namespace": "openshift-etcd",
          "Name": "etcd-peer-\u003cbootstrap\u003e"
        },
        "certKeyInfo": {
          "selectedCertMetadataAnnotations": [
            {
              "key": "openshift.io/owning-component",
              "value": "etcd"
            },
            {
              "key": "openshift.io/description",
              "value": "Peer Cert for node \u003cbootstrap\u003e"
            }
          ],
          "owningJiraComponent": "etcd",
          "description": "Peer Cert for node \u003cbootstrap\u003e"
        }
      },
```

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go [hunk 1]
```diff
@@ -161,34 +162,38 @@ func skipRevisionedInOnDiskLocation(location certgraphapi.OnDiskLocation) bool {
...
-	nodes := map[string]int{}
+func rewriteSecretDetails(secret *corev1.Secret, original, replacement string) {
+	name := strings.ReplaceAll(secret.Name, original, replacement)
+	if secret.Name != name {
+		secret.Name = name
+		if len(secret.Annotations) == 0 {
+			secret.Annotations = map[string]string{}
+		}
+		// Replace node name from annotation value
+		for key, value := range secret.Annotations {
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:173]: ok for now, but I suspect we may need to do keys too.

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go [hunk 2]
```diff
@@ -199,6 +204,31 @@ func RewriteNodeIPs(nodeList []*corev1.Node) *metadataOptions {
 	}
 }
 
+func RewriteBootstrapIPs(bootstrapIP string) *metadataOptions {
+	return &metadataOptions{
+		rewriteSecretFn: func(secret *corev1.Secret) {
+			if secret.Namespace != "openshift-etcd" {
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:210]: unnecessary restriction.

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go [hunk 3]
```diff
@@ -199,6 +204,31 @@ func RewriteNodeIPs(nodeList []*corev1.Node) *metadataOptions {
...
+				return
+			}
+
+			certHostNames, ok := secret.Annotations["auth.openshift.io/certificate-hostnames"]
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:214]: I suggest doing a replace of `descriptionWords[len(descriptionWords)-1]` on all secrets by first doing the list of etcd secrets and locating the name .  I'm willing to merge this as a starting point, but I'd like to see a refactor that identifies the bootstrap host name and replaces that.

**Comment by deads2k** [comment]
/lgtm
/approve
/hold

holding for acknowledgement of follow-ups.  Feel free to release once acknowledged.

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go (hunk 2 above)
Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:210]: Other components may use `auth.openshift.io/certificate-hostnames` annotation differently, perhaps we should rename `RewriteBootstrapIPsInEtcdSecrets`?

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go (hunk 1 above)
Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:173]: Right, agreed

File: pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go (hunk 3 above)
Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go:214]: Right, I think its best to have `getBootstrapHostnameByIP` function which does that

**Comment by vrutkovs** [comment]
Agree on followups
/hold cancel

**Pull request merged**